src/System.ComponentModel.TypeConverter/tests/GuidConverterTests.cs

```diff
@@ -10,89 +10,42 @@
-        [InlineData("invalid")]
-        public void ConvertFrom_InvalidString_ThrowsFormatException(string value)
-        {
-            var converter = new GuidConverter();
-            Assert.Throws<FormatException>(() => converter.ConvertFrom(value));
-        }
-
-        [Theory]
-        [InlineData(null)]
-        [InlineData(1)]
-        public void ConvertFrom_InvalidValue_ThrowsNotSupportedException(object value)
-        {
-            var converter = new GuidConverter();
-            Assert.Throws<NotSupportedException>(() => converter.ConvertFrom(value));
-        }
-
-        [Theory]
-        [InlineData(typeof(string), true)]
-        [InlineData(typeof(InstanceDescriptor), true)]
-        [InlineData(typeof(Guid), false)]
-        [InlineData(null, false)]
-        public void CanConvertTo_Invoke_ReturnsExpected(Type destinationType, bool expected)
-        {
-            var converter = new GuidConverter();
-            Assert.Equal(expected, converter.CanConvertTo(destinationType));
-        }
-
-        [Fact]
-        public void ConvertTo_String_ReturnsExpected()
-        {
-            var converter = new GuidConverter();
-            var value = new Guid(0x30da92c0, 0x23e8, 0x42a0, 0xae, 0x7c, 0x73, 0x4a, 0x0e, 0x5d, 0x27, 0x82);
-            Assert.Equal("30da92c0-23e8-42a0-ae7c-734a0e5d2782", converter.ConvertTo(value, typeof(string)));
+            yield return ConvertTest.Throws<FormatException>("");
+            yield return ConvertTest.Throws<FormatException>("invalid");
+            yield return ConvertTest.CantConvertFrom(1);
+            yield return ConvertTest.CantConvertFrom(new object());
         }
 
-        [Fact]
-        public void ConvertTo_InstanceDescriptor_ReturnsExpected()
+        public override IEnumerable<ConvertTest> ConvertToTestData()
         {
-            var converter = new GuidConverter();
-            var value = new Guid(0x30da92c0, 0x23e8, 0x42a0, 0xae, 0x7c, 0x73, 0x4a, 0x0e, 0x5d, 0x27, 0x82);
-            InstanceDescriptor descriptor = Assert.IsType<InstanceDescriptor>(converter.ConvertTo(value, typeof(InstanceDescriptor)));
-            ConstructorInfo constructor = Assert.IsAssignableFrom<ConstructorInfo>(descriptor.MemberInfo);
-            Assert.Equal(new Type[] { typeof(string) }, constructor.GetParameters().Select(p => p.ParameterType));
-            Assert.Equal(new object[] { "30da92c0-23e8-42a0-ae7c-734a0e5d2782" }, descriptor.Arguments);
+            yield return ConvertTest.Valid(new Guid(0x30da92c0, 0x23e8, 0x42a0, 0xae, 0x7c, 0x73, 0x4a, 0x0e, 0x5d, 0x27, 0x82), "30da92c0-23e8-42a0-ae7c-734a0e5d2782");
+            yield return ConvertTest.Valid(
+                new Guid(0x30da92c0, 0x23e8, 0x42a0, 0xae, 0x7c, 0x73, 0x4a, 0x0e, 0x5d, 0x27, 0x82),
+                new InstanceDescriptor(
+                    typeof(Guid).GetConstructor(new Type[] { typeof(string) }),
+                    new object[] { "30da92c0-23e8-42a0-ae7c-734a0e5d2782" }
```

Comment:
nit: 30da92c0-23e8-42a0-ae7c-734a0e5d2782 is used multiple times in the file, you can save it as a variable.